tests/2015-11-01/Microsoft.Network.tests.json

```diff
@@ -0,0 +1,262 @@
+        },
+        {
+          "message": "Missing required property: monitorConfig",
+          "dataPath": "/properties"
+        }
+      ],
+      "definition": "http://schema.management.azure.com/schemas/2015-11-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles",
+      "json": {
+        "name": "TestProfile-1",
+        "type": "Microsoft.Network/trafficManagerProfiles",
+        "apiVersion": "2015-11-01",
+        "properties": { }
+      }
+    },
+    {
+      "name": "Traffic Manager profile with missing required properties (level 2)",
+      "expectedErrors": [
+        {
+          "message": "Missing required property: relativeName",
+          "dataPath": "/properties/dnsConfig"
+        },
+        {
+          "message": "Missing required property: ttl",
+          "dataPath": "/properties/dnsConfig"
+        },
+        {
+          "message": "Missing required property: protocol",
+          "dataPath": "/properties/monitorConfig"
+        },
+        {
+          "message": "Missing required property: port",
+          "dataPath": "/properties/monitorConfig"
+        },
+        {
+          "message": "Missing required property: path",
+          "dataPath": "/properties/monitorConfig"
+        },
+        {
+          "message": "Missing required property: name",
+          "dataPath": "/properties/endpoints/0"
+        },
+        {
+          "message": "Missing required property: type",
+          "dataPath": "/properties/endpoints/0"
+        },
+        {
+          "message": "Missing required property: properties",
+          "dataPath": "/properties/endpoints/0"
+        }
+      ],
+      "definition": "http://schema.management.azure.com/schemas/2015-11-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles",
+      "json": {
+        "name": "TestProfile-1",
+        "type": "Microsoft.Network/trafficManagerProfiles",
+        "apiVersion": "2015-11-01",
+        "properties": {
+          "profileStatus": "Enabled",
+          "trafficRoutingMethod": "Priority",
+          "dnsConfig": { },
+          "monitorConfig": { },
+          "endpoints": [ { } ]
+        }
+      }
+    },
+    {
+      "name": "Traffic Manager profile with invalid property values",
+      "expectedErrors": [
+        {
+          "message": "No enum match for: \"gibberish\"",
+          "dataPath": "/type"
+        },
+        {
+          "message": "No enum match for: \"gibberish\"",
+          "dataPath": "/apiVersion"
+        },
+        {
+          "message": "No enum match for: \"gibberish\"",
+          "dataPath": "/properties/endpoints/0/type"
+        }
+      ],
+      "definition": "http://schema.management.azure.com/schemas/2015-11-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles",
+      "json": {
+        "name": "TestProfile-1",
+        "type": "gibberish",
+        "apiVersion": "gibberish",
+        "properties": {
+          "profileStatus": "gibberish",
```

Comment:
It doesn't, though, for the same reason that it didn't for anyOf: "gibberish" doesn't satisfy the enum branch but is considered an expression so no failure